docs/src/lens.md

```diff
@@ -1,149 +1,251 @@
 # Lenses and the `Lens` trait
 
-Let's say we're building a todo list application, and we are designing the widget
-that will represent a single todo item. Our data model looks like this:
+One of the key abstractions in `druid` along with `Data` is the `Lens` trait. This page explains what they are, and then how to use them. `Lens`es may seem complicated at first, but they are also very powerful, allowing you to write code that is reusable, concise, and understandable (once you understand `Lens`es themselves).
```

Comment:
We've decided on Druid as the standard spelling for the project, only doing `druid` if we're referring to the crate (as in, talking about `druid` vs `druid-shell`.

I'd also be fine with 'Lenses' once we're talking about the concept, and not the trait. 🤓